packages/styles/table.css

```diff
@@ -30,6 +36,44 @@
   border-bottom: 2px solid var(--table-row-border-color);
 }
 
+.TableHeader .sort-button {
+  background-color: transparent;
+  width: 100%;
+  /* a fixed height prevent TableHeader from being jumpy */
+  height: 1.625rem;
```

Comment:
I see you added a fixed height to fix the header from jumping, but it seems like something else may be going on behind the scenes that may need more investigating (such as the table header briefly rendering without a button causing the height to be different). It would be nice to try to determine the root cause to see if we're not rendering something correctly.

Reply:
Turns out that sometimes there is a brief moment when no Icon is rendered yet. I set the span's height to the height of an Icon to prevent the brief moment changes the height of the button.

Reply:
It looks as though the entire table is re-mounting (the HTML elements are being totally replaced). That shouldn't be necessary for all of the elements, especially the non-sorting headers. React should be able to tell that some of this doesn't need to change. There also shouldn't be a lag in rendering the Icon. I agree with @scurker that I'm not sure CSS is the right fix here. I'd be happy to jump on a call and look at this together with you.

Reply:
I think you hit on something @thuey, it's probably re-rendering the icon when the sort value changes. I wonder if we should handle this differently here since the browser has to perform a layout calculation each time the svg renders - there's probably an optimization we can make here to ensure that doesn't happen. Have you tried wrapping the icon with `React.memo`?

Reply:
Let's jump on a call tomorrow and discuss it.

Reply:
We realized that the Demo component causes extra renders (usually for complex component). Therefore in the following changes we manually set up the demo.